server/service/src/invoice/inbound_shipment/delete/validate.rs

```diff
@@ -44,3 +46,9 @@ impl From<InvoiceLinesExist> for DeleteInboundShipmentError {
         DeleteInboundShipmentError::InvoiceLinesExists(error.0)
     }
 }
+
+impl From<NotThisStoreInvoice> for DeleteInboundShipmentError {
```

Comment:
Just fyi, we've been using a different pattern from the original 'From', which is mapping error if it's caught, for example in this one:
```rust
pub fn validate(
    connection: &StorageConnection,
    store_id: &str,
    patch: &UpdateOutboundShipment,
) -> Result<(InvoiceRow, Option<Name>), UpdateOutboundShipmentError> {
    use UpdateOutboundShipmentError::*;
    let invoice = check_invoice_exists(&patch.id, connection)?;
    check_store(&invoice, store_id)?;
    check_invoice_type(&invoice)?;
    check_invoice_is_editable(&invoice)?;
    check_invoice_status(&invoice, patch.full_status(), &patch.on_hold)?;
    check_can_change_status_to_allocated(connection, &invoice, patch.full_status())?;

    let other_party_id = match &patch.other_party_id {
        None => return Ok((invoice, None)),
        Some(other_party_id) => other_party_id,
    };

    let other_party = check_other_party(
        connection,
        store_id,
        &other_party_id,
        CheckOtherPartyType::Customer,
    )
    .map_err(|e| match e {
        OtherPartyErrors::OtherPartyDoesNotExist => OtherPartyDoesNotExist {},
        OtherPartyErrors::OtherPartyNotVisible => OtherPartyNotVisible,
        OtherPartyErrors::TypeMismatched => OtherPartyNotACustomer,
        OtherPartyErrors::DatabaseError(repository_error) => DatabaseError(repository_error),
    })?;

    Ok((invoice, Some(other_party)))
}
```

When check_other_party was added, explicit map_err was used rather than from, we thought that `From` was a bit of an indirection, and in this case it was better to inline the mapping.

Reply:
Btw the pattern above wasn't really communicated that well sorry. What you've done is good. If i was doing it i would:
```rust
if invoice.store_id != store_id {
  return Err(Error::NotThisStoreInvoice)
}
```